### DISCUSSION TITLE
Dreame L10 Pro lost network and serial console

### DISCUSSION THREAD

**Discussion by wailnucleus** [Hypfer/Valetudo#1780] [Q&A / Support] [answered]
Hello

Yesterday, my Dreame L10 Pro suddenly lost it's network connection. No changes have been applied to the robot since at least some weeks.

I can switch it on and off, the startup and shutdown sounds are played and it cleans the floor via button press, but it does not connect itself to the WLAN anymore.

I also tried to connect to it via serial console, but without any success (I always get a blank screen, even after swapping RX and TX and using another, completely different UART adapter). The installation one year ago worked flawlessly.

My idea would be to reset and re-configure the WI-FI by pressing "Spot Cleaning" and "Home" buttons.
Is there anything else I can do before to better understand the issue?

**Comment by Hypfer** [comment]
> (I always get a blank screen, even after swapping RX and TX and using another, completely different UART adapter)

Nah. Can't be. That's just a cabling issue or something like that. The UART doesn't go away in any situation.
You should still have SSH access though so it shouldn't matter.

> My idea would be to reset and re-configure the WI-FI by pressing "Spot Cleaning" and "Home" buttons.

That probably won't work as there likely is no Valetudo on `/data` anymore.

What you've experienced unfortunately happens from time to time. If you run a recent dustbuilder firmware build, you should also have fixes for some of the causes for that. However, we can't fix all of them.
What likely happend is that the firmware was not able to mount the `/data` partition because of filesystem corruption.
In that case, it creates a new FS which wipes Valetudo etc.

You can verify that that was the cause by running `PKG_EXECPATH=PKG_INVOKE_NODEJS /data/valetudo` to get an interactive js shell and then pasting the following:

```js
let fd = fs.openSync("/dev/by-name/UDISK", "r");
let buf = Buffer.alloc(1024)
fs.readSync(fd, buf, {position: 1024, length: 1024});
fs.closeSync(fd);
console.log({mntCount: buf.readUInt16LE(52), totalMiBWritten: Number(buf.readBigInt64LE(376) / BigInt(1024))})
```

If mount count and totalMiBWritten are fairly low numbers, it's a freshly created ext4 fs. Note that you will have to reinstall Valetudo first.


To recover from this, just follow the second half of the rooting instructions again. I'd recommend flashing a new dustbuilder firmware package with the prepackage Valetudo option selected. As mentioned, you should still have SSH access and thus won't need the UART.

**Comment by wailnucleus** [comment]
Thank you very much for your quick reply and support.

Unfortunately, I do not have ssh access as the robot does not connect to the WLAN router. There are no leases visible on the router for the robot and I also do not see the WLAN light of the robot flashing when powering up.

Today I tried again to access the robot via serial console, but without success.

Anything else I can do to improve the situation?

Thank you and kind regards

**Reply by Hypfer** [reply]
The Wi-Fi configuration is also stored on the `/data` partition that has most likely been wiped.
You need to connect to the Wi-Fi AP of the robot again.

Note that the AP gets disabled 30 minutes or so after bootup. If you don't see it, press and hold the two outer buttons until it talks to you or just power cycle the robot.

**Reply by wailnucleus** [reply, marked as answer]
Thank you, very much appreciated. I will try that and report back.

Kind regards

**Comment by wailnucleus** [comment]
1. Reset the WiFi by pressing and holding the two outer buttons
2. Connected notebook to the now visible Dreame WLAN Access Point
3. ssh 192.168.5.1 -lroot

Et voilà:
--------------------------------------------------------------
    _   _   _                        _     _                  
   / \ | |_| |__   ___ _ __   __ _  | |   (_)_ __  _   ___  __
  / _ \| __| '_ \ / _ \ '_ \ / _` | | |   | | '_ \| | | \ \/ /
 / ___ \ |_| | | |  __/ | | | (_| | | |___| | | | | |_| |>  < 
/_/   \_\__|_| |_|\___|_| |_|\__,_| |_____|_|_| |_|\__,_/_/\_\
                                                              
--------------------------------------------------------------
 Athena Linux (p2029_release)
--------------------------------------------------------------

build with dustbuilder (https://builder.dontvacuum.me)
Sat 22 Jan 2022 06:24:03 PM UTC
1105
...

It worked, will do the rest during the weekend!
Thank you very much again!

Kind regards

**Comment by wailnucleus** [comment]
Thank you very much!
